dockstore-webservice/src/main/java/io/dockstore/webservice/core/LimitedSourceFileBuilder.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.core;
+
+import com.google.common.primitives.Bytes;
+import io.dockstore.common.DescriptorLanguage;
+import java.nio.charset.StandardCharsets;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Build a SourceFile to which we've applied Dockstore's content limitations.
+ * Require the file type, content, and paths.
+ */
+public class LimitedSourceFileBuilder {
+
+    private static final long BYTES_PER_MEGABYTE = 1024L * 1024L;
+    private static final long MAXIMUM_FILE_SIZE = BYTES_PER_MEGABYTE;
+    private static final long NOTEBOOK_MAXIMUM_FILE_SIZE = 3 * BYTES_PER_MEGABYTE;
```

Comment:
These are the content-specific file size limits.  The 10MB limit is a download limit, the GitHub code will just not download files over that size.  It would be nice to apply the limits in one place, but the current structure of the code makes that difficult.